app/models/course_template.rb

```diff
@@ -37,4 +43,35 @@ def valid_source_url?
   def clonable?
     !hidden && (expires_at.nil? || expires_at > Time.now)
   end
+
+  def cache_path
+    "#{Course.cache_root}/#{name}-#{cache_version}"
+  end
+
+  def refresh
+    firstcourse = true
+    courses.each do |c|
+      c.refresh no_directory_changes: !firstcourse
+      reload
+      firstcourse = false
+    end
```

Comment:
Hmm, when thinking of this, cleaner way could be that every course would be based on course template (even the old ones and single custom courses). Then all the refresh logic for repo and file handling could be made always for template only. 

But I think it takes quite big refactoring of whole refresh system... Not sure if it would be wiser to implement later on different task. Even though it was discussed earlier too elsewhere.